tools-local/Microsoft.ML.CodeAnalyzer/WhitespaceAnalyzer.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.ML.CodeAnalyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class WhitespaceAnalyzer : DiagnosticAnalyzer
```

Comment:
:bulb: Consider using one of the maintained alternatives instead of roll-your-own analyzer for this.

Reply:
Great. Point me at one?

Reply:
[StyleCop.Analyzer](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/) I see. Let's see if I can get that to work...

Reply:
Replaced with SC rules [SA1028](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1028.md) and [SA1507](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1507.md). Incidentally this change is accompanied by many changes to files because this also covers XML doc comments.